controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1740,6 +1758,33 @@ func saveHostProvisioningSettings(host *metal3api.BareMetalHost, info *reconcile
 	return dirty, nil
 }
 
+func (r *BareMetalHostReconciler) saveHostFirmwareComponents(prov provisioner.Provisioner, info *reconcileInfo, hfc *metal3api.HostFirmwareComponents) (dirty bool, err error) {
+	dirty = false
+	if !reflect.DeepEqual(hfc.Status.Updates, hfc.Spec.Updates) {
+		info.log.Info("HostFirmwareComponents Status", "updates", hfc.Status.Updates)
+
+		for i := range hfc.Spec.Updates {
+			hfc.Spec.Updates[i].DeepCopyInto(&hfc.Status.Updates[i])
+		}
+
+		info.log.Info("re assigned status updates")
+		info.log.Info("New HostFirmwareComponents Status", "updates", hfc.Status.Updates)
+
+		// Retrieve new information about the firmware components stored in ironic
+		components, err := prov.GetFirmwareComponents()
+		if err != nil {
+			info.log.Info("Failed to get new information for firmware components in ironic")
+			return dirty, err
+		}
+		hfc.Status.Components = components
```

Comment:
Is there a reason we are setting this in two different controllers?

Reply:
yes, if there is a firmware update it would be executed in the `actionPreparing` , we call `saveHostFirmwareComponents` later on, when a firmware update is executed new information is available in Ironic and we need the retrieve it, so we call `prov.GetFirmwareComponents()` and we re-assign the `Status.Components` to the ones we just retrieved.

Reply:
I understand what it's doing; I'm not clear on why it needs to do it in two different places. Does it need to be atomic with the Updates update?

Reply:
I have the feeling it should 😅  @dtantsur  @honza  do you have some thoughts ?  Maybe I can just move a few things to be handle here and not in the HFC Controller.. 🤔 
* [BMH Controller](https://github.com/openshift/baremetal-operator/pull/355/files/fe2e66774bbdcdd0ed11e5d51efd697039b544f6#diff-a69d9029388ab766ed36b32180145f52785a9d4a153775510dbddfa928a72e1cR1779) 
* [HFC Controller](https://github.com/openshift/baremetal-operator/pull/355/files/fe2e66774bbdcdd0ed11e5d51efd697039b544f6#diff-bdd82eab1aa7657521e01e0ec4b794bfd09e1b4c287a539f5f4366ec3a321c58R196)

Reply:
I believe for BIOS settings, we only set `status.settings` in the HFS controller.

There is a bit of a risk in this approach though: I've seen cases where a race between BMH and HFS controllers caused the process to start the 2nd time (only happens in servicing, probably not an issue for the currently present code).

Reply:
> Maybe I can just move a few things to be handle here and not in the HFC Controller.

Here only runs during the Preparing state, so it's not obvious to me that doing it here would be correct.

Reply:
Preparing should be enough, since the data only changes when manual cleaning is executed